xelerance/Openswan
# T8419 travis validate libpluto

Refactoring of TravisCI test matrix, adding of validate-libpluto test, and a few minor changes to 'run-unit-tests.sh' script.

Changes to TravisCI config...
- TravisCI 'matrix' uses "TEST_JOB" variable to designate jobs
- TravisCI 'script' runs tests/utils/ shell scripts

There are three configured jobs...
- build-and-check - runs make programs and make check
- build-every-version - runs make programs on all commits since last tag
- validate-libpluto - runs 'run-unit-tests.sh' in update pcap mode, to
                      catch inconsistencies
